packages/client/test/sync/fetcher/blockfetcher.spec.ts

```diff
@@ -101,6 +102,36 @@ tape('[BlockFetcher]', async (t) => {
     t.end()
   })
 
+  t.test('store()', async (st) => {
+    td.reset()
+    st.plan(2)
+
+    const config = new Config({ maxPerRequest: 5, transports: [] })
+    const pool = new PeerPool() as any
+    const chain = new Chain({ config })
+    chain.putBlocks = td.func<any>()
+    const fetcher = new BlockFetcher({
+      config,
+      pool,
+      chain,
+      first: new BN(1),
+      count: new BN(10),
+      timeout: 5,
+    })
+    td.when(chain.putBlocks(td.matchers.anything())).thenReject(new Error('err0'))
+    try {
+      await fetcher.store([])
+    } catch (err: any) {
+      st.ok(err.message === 'err0', 'store() threw on invalid block')
+    }
+    td.reset()
+    chain.putBlocks = td.func<any>()
+    td.when(chain.putBlocks(td.matchers.anything())).thenResolve(1)
+    config.events.on(Event.SYNC_FETCHER_FETCHED, () =>
+      st.pass('store() emitted SYNC_FETCHER_FETCHED event on putting blocks')
+    )
+    await fetcher.store([])
+  })
```

Comment:
Nice, all these new test cases are really great as well! 😄 

Especially the Fetcher is a beasty thing, so that's nice if we get to better coverage there, also for reproducing things and having certain functionality parts more directly called.